tests/test_bashcomplete.py

```diff
@@ -407,3 +407,45 @@ def hsub():
     # If the user exactly types out the hidden command, complete its subcommands.
     assert choices_without_help(cli, ['hgroup'], '') == ['hgroupsub']
     assert choices_without_help(cli, ['hsub'], '--h') == ['--hname']
+
+
+def test_args_with_double_dash():
+    def _complete(ctx, args, incomplete):
+        _tmp =  ['abc', 'def', 'ghi', 'jkl', 'mno', 'pqr', 'stu', 'vwx', 'yz']
+        _tmp += ['-abc', '--def', '-ghi', '--jkl', '-osk', '--oab']
+        return [x for x in _tmp if x.startswith(incomplete)]
+
+    @click.group()
+    def entrypoint():
+        pass
+
+    @click.command()
+    @click.option('--opt')
+    @click.argument('arg', nargs=-1, autocompletion=_complete)
+    def subcommand(opt, arg):
+        pass
+
+    entrypoint.add_command(subcommand)
+
+    assert choices_without_help(entrypoint, ['subcommand'], '-') == ['--opt']
+    assert choices_without_help(entrypoint, ['subcommand'], '-o') == []
+    assert choices_without_help(entrypoint, ['subcommand', 'abc'], '') == _complete(0,0,'')
+    assert choices_without_help(entrypoint, ['subcommand', 'abc'], 'a') == _complete(0,0,'a')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '') == _complete(0,0,'')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], 'a') == _complete(0,0,'a')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '-') == _complete(0,0,'-')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '--') == _complete(0,0,'--')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '-g') == _complete(0,0,'-g')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '-o') == _complete(0,0,'-o')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '--o') == _complete(0,0,'--o')
+    assert choices_without_help(entrypoint, ['subcommand', '--'], '---') == _complete(0,0,'---')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', 'abc'], '') == _complete(0,0,'')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', 'abc'], 'a') == _complete(0,0,'a')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '') == _complete(0,0,'')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], 'a') == _complete(0,0,'a')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '-') == _complete(0,0,'-')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '--') == _complete(0,0,'--')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '-g') == _complete(0,0,'-g')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '-o') == _complete(0,0,'-o')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '--o') == _complete(0,0,'--o')
+    assert choices_without_help(entrypoint, ['subcommand', '--opt', '--'], '---') == _complete(0,0,'---')
```

Comment:
Use `pytest.mark.parametrize` instead of writing out individual `assert` statements.